katharsis-core/src/main/java/io/katharsis/queryspec/QuerySpec.java

```diff
@@ -0,0 +1,169 @@
+package io.katharsis.queryspec;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.katharsis.utils.CompareUtils;
+
+public class QuerySpec {
+
+	private Class<?> resourceClass;
+	private Long limit = null;
+	private long offset = 0;
+	private List<FilterSpec> filters = new ArrayList<>();
+	private List<SortSpec> sort = new ArrayList<>();
+	private List<IncludeFieldSpec> includedFields = new ArrayList<>();
+	private List<IncludeRelationSpec> includedRelations = new ArrayList<>();
+	private Map<Class<?>, QuerySpec> relatedSpecs = new HashMap<>();
+
+	public QuerySpec(Class<?> resourceClass) {
+		this.resourceClass = resourceClass;
+	}
+
+	public Class<?> getResourceClass() {
+		return resourceClass;
+	}
+
+	/**
+	 * Evaluates this querySpec against the provided list in memory. It supports
+	 * sorting, filter and paging. 
+	 * 
+	 * TODO currently ignores relations and inclusions, has room for improvements
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)